presto-orc/src/test/java/com/facebook/presto/orc/TestDwrfDictionaries.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc;
+
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.PageSourceOptions;
+import com.facebook.presto.spi.PageSourceOptions.FilterFunction;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.VarcharType;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.units.DataSize;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static com.facebook.presto.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static com.facebook.presto.orc.OrcReader.MAX_BATCH_SIZE;
+import static com.facebook.presto.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static com.google.common.io.Resources.getResource;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static org.testng.Assert.assertEquals;
+
+public class TestDwrfDictionaries
+{
+    @Test
+    public void testSlice()
+            throws Exception
+    {
+        Block block1 = readFile("test_dictionaries/string-dictionary.orc", VarcharType.VARCHAR);
```

Comment:
- block1 and block2 are confusing as it is not clear which is being tested. I'd rename these to expected and actual.
- I'd also rename compare method to assertEqualBlock
- I'd also inline null1 and null2 variables

```
    @Test
    public void testSlice()
            throws Exception
    {
        Block expected = readFile("test_dictionaries/string-dictionary.orc", VARCHAR);
        Block actual = readFile("test_dictionaries/string-row-group-dictionary.dwrf", VARCHAR);
        assertEqualBlocks(VARCHAR, expected, actual);
    }

    private static void assertEqualBlocks(Type type, Block expected, Block actual)
    {
        assertEquals(expected.getPositionCount(), actual.getPositionCount());
        for (int i = 0; i < expected.getPositionCount(); i++) {
            assertEquals(actual.isNull(i), expected.isNull(i));
            if (!expected.isNull(i)) {
                assertEquals(type.compareTo(expected, i, actual, i), 0);
            }
        }
    }
```